### ISSUE TITLE
Handling of generic classes throws NPE

### ISSUE THREAD

**Issue by aurae** [bluelinelabs/LoganSquare#99]
I'm maintaining a Retrofit converter for LoganSquare. In the process of updating to the latest version `1.2.1` of LoganSquare, I wrote new test cases to check if the new generics feature would work with the Converter. However, I noticed that LoganSquare's serialization and unserialization with generics doesn't work at all for me, even in an isolated test project.

The error thrown upon running the unit test is:
```
java.lang.RuntimeException: JsonMapperLoaderImpl class not found. Have you included the steps needed for LoganSquare to process your annotations?
	at com.bluelinelabs.logansquare.LoganSquare.<clinit>(LoganSquare.java:31)
	... 35 more
```

I dug into the issue, and this error message isn't accurate for the case at hand, though. The JsonMapperLoaderImpl is properly generated and found on runtime (as shown by `Class.forName()`). Instead, the error happens in [LoganSquare's `mapperFor` method, line 247](https://github.com/bluelinelabs/LoganSquare/blob/master/core/src/main/java/com/bluelinelabs/logansquare/LoganSquare.java#L247), because it can't find a mapper for the generic implementation I'm passing in, which throws an NPE upon trying to insert the null value into the `PARAMETRIZED_OBJECT_MAPPERS` Map.

I have created [a gist](https://gist.github.com/aurae/d6f27698d9d83d7296cb) containing my setup.

**Comment by EricKuck** [comment]
Man, you're quick! I haven't even had time to document the new stuff yet :)

So the new `ParameterizedType` method of parsing is only for classes that are themselves parameterized, not for classes that have parameterized super classes. Line 12 of your code can change to 

`String json = LoganSquare.serialize(model, Model.class);`

and line 21 can change to 

`Model model = LoganSquare.parse(json, Model.class);`

If you want to test the `ParameterizedType` method, try adding something like this to your project:

```
@JsonObject
public class GenericModel<T> extends AbstractModel<T> {
    @JsonField
    double extra;
}
```

```
    @Test public void testSerialization() throws Exception {
        GenericModel<String> model = new GenericModel<>();
        model.name = "The Name";
        model.extra = 2.5;
        model.value = 2;
        model.content = "String Content";

        // Throws at this line
        String json = LoganSquare.serialize(model, new ParameterizedType<GenericModel<String>>() { });
        Assert.assertEquals(true, true);
    }

    @Test public void testParsing() throws Exception {
        String json = "{\"name\":\"The Name\",\"value\":1,\"extra\":2.5,\"content\":\"String content\"}";

        // Throws at this line
        GenericModel<String> model = LoganSquare.parse(json, new ParameterizedType<GenericModel<String>>() { });
        Assert.assertEquals(true, true);
    }
```

Should have foreseen this being an issue, so I'll add checks that make it work the way you'd expect (or at least provide better error messaging) by the time the docs go up.

PS - Thanks for maintaining the Retrofit converter. Very much appreciated!

**Comment by aurae** [comment]
Oh, that makes sense. I will check it out momentarily!

**Comment by EricKuck** [comment]
I take that back. Looks like there is an issue with abstract generic classes. I'm working on a fix for this now. In the meantime, you can fix the issue you have by declaring that class non-abstract.

**Comment by EricKuck** [comment]
OK this is actually fixed now in v1.2.2. I just pushed it to JCenter, so it might take a minute to become available.

**Comment by aurae** [comment]
It works like a charm now! Thanks :).
